docs/markdown/FAQ.md

```diff
@@ -683,3 +683,21 @@ executable(
   link_language : 'c',
 )
 ```
+
+## How do I use `include-what-you-use` with Meson?
+
+Following the steps from
+[Using with a compilation database](https://github.com/include-what-you-use/include-what-you-use#using-with-a-compilation-database),
+configure the build directory (e.g., with `meson setup build`), and run:
+
+```console
+$ iwyu-tool -p build
+```
+
+See `iwyu-tool --help` for more options.
+
+A run target can also be added with the following meson code:
+```meson
+iwyu_tool = find_program('iwyu-tool', 'iwyu_tool')
+run_target('iwyu', command: [iwyu_tool, '-p', meson.project_build_root()])
```

Comment:
This can't be the project build root, as that will be wrong inside a subproject

Reply:
I don't think run_targets() get generated for subprojects, so it doesn't really matter. I have never really gotten confirmation on this though

Reply:
There's `meson.build_root()`. But maybe it would be better to have an ifdef so the target is only created if the current project is the toplevel project (which is what most people probably expect).

Reply:
If memory serves, subproject run targets are generated as `subprojname:targetname`.

Reply:
*meson.global_build_root() probably

Reply:
```console
$ ninja -C build/ hse:shellcheck
ninja: Entering directory `build/'
ninja: error: unknown target 'hse:shellcheck', did you mean 'hse@@shellcheck'?
```
That `@@` is gross. Is that right?

Reply:
It's an illegal filename character, and this was the rationale in commit 02bea7d5bf5c98c586e59e60357fc5ca11826172.

Granted, a run_target isn't a "real" filename, as ninja never expects to actually *create* it... but it is written to build.ninja as though it were one.

`meson test` doesn't even pretend to take filenames.

Reply:
> It's an illegal filename character

Never knew this. Good to know then.

Reply:
(Technically it's not an illegal filename character, it's a *sometimes* illegal one. It's illegal on Windows, but not Unix.)